lib/galaxy/datatypes/interval.py

```diff
@@ -167,40 +167,39 @@ def get_estimated_display_viewport(self, dataset, chrom_col=None, start_col=None
             start = sys.maxsize
             end = 0
             max_col = max(chrom_col, start_col, end_col)
-            fh = open(dataset.file_name)
-            while True:
-                line = fh.readline(VIEWPORT_READLINE_BUFFER_SIZE)
-                # Stop if at end of file
-                if not line:
-                    break
-                # Skip comment lines
-                if not line.startswith('#'):
-                    try:
-                        fields = line.rstrip().split('\t')
-                        if len(fields) > max_col:
-                            if chrom is None or chrom == fields[chrom_col]:
-                                start = min(start, int(fields[start_col]))
-                                end = max(end, int(fields[end_col]))
-                                # Set chrom last, in case start and end are not integers
-                                chrom = fields[chrom_col]
-                            viewport_feature_count -= 1
-                    except Exception:
-                        # Most likely a non-integer field has been encountered
-                        # for start / stop. Just ignore and make sure we finish
-                        # reading the line and decrementing the counters.
-                        pass
-                # Make sure we are at the next new line
-                readline_count = VIEWPORT_MAX_READS_PER_LINE
-                while line.rstrip('\n\r') == line:
-                    assert readline_count > 0, Exception('Viewport readline count exceeded for dataset %s.' % dataset.id)
-                    line = fh.readline(VIEWPORT_READLINE_BUFFER_SIZE)
+            with open(dataset.file_name) as fh:
+                for line in util.iter_start_of_line(fh, VIEWPORT_READLINE_BUFFER_SIZE):
+                    # Stop if at end of file
                     if not line:
```

Comment:
Is this check still needed, given you are using a for loop?

Reply:
it's not, I dropped it in the other functions and forgot this one, thanks!